pkg/controllers/controller.go

```diff
@@ -49,18 +50,22 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 	}
 	// 2. Copy object for merge patch base
 	persisted := resource.DeepCopyObject()
-	// 3. Set to true to remove race condition
+	// 3. Set Defaults
+	if _, ok := resource.(apis.Defaultable); ok {
+		resource.(apis.Defaultable).SetDefaults(ctx)
+	}
+	// 4. Set to true to remove race condition
 	// TODO: remove status conditions on provisioners
 	if provisioner, ok := resource.(*v1alpha1.Provisioner); ok {
 		provisioner.StatusConditions().MarkTrue(v1alpha1.Active)
 	}
-	// 4. Reconcile
+	// 5. Reconcile
 	if _, err := c.Controller.Reconcile(ctx, resource); err != nil {
 		zap.S().Errorf("Controller failed to reconcile kind %s, %s",
 			resource.GetObjectKind().GroupVersionKind().Kind, err.Error())
 		return reconcile.Result{Requeue: true}, nil
 	}
-	// 5. Update Status using a merge patch
+	// 6. Update Status using a merge patch
 	if !reflect.DeepEqual(resource, persisted) {
```

Comment:
Unrelated to your PR, but I realized the other day that we should be comparing statuses and patching statuses, or comparing objects and patching objects. This is currently mismatched: we compare the objects, but only patch the statuses. If resource has a changed spec, we'll send an empty patch.

Reply:
I remember talking about that. I'll change it here.

Reply:
Additionally, if we only update the status, this actually won't call the patch. I think until we have the statuses working, we should delete this line.

Reply:
SGTM -- can you own this?